pages/community/create.tsx

```diff
@@ -0,0 +1,319 @@
+} from 'governator-sdk';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  useFieldArray,
+  useForm,
+} from 'react-hook-form';
+import * as yup from 'yup';
+import { useSession } from 'next-auth/react';
+import { discordAxios, privateBaseAxios, privateBaseFetcher } from 'constants/axios';
+import React, { useCallback, useState, useEffect } from 'react';
+import { FiMinus, FiPlus } from 'react-icons/fi';
+
+
+const schema = yup.object({
+  name: yup.string().required('Community name is required.'),
+  administrators: yup
+    .array<{
+      user_allowlist: Array<number>;
+      provider_id: string;
+    }>()
+    .required('At least one administrator is required.'),
+  client_config: yup
+    .array<{
+      provider_id: string;
+      guild_id?: string;
+    }>()
+    .required('Client config is required.'),
+}).required();
+
+interface CommunityForm {
+  name: string;
+  administrators: Array<{
+    user_allowlist: Array<number>;
+    provider_id: string;
+  }>;
+  client_config: Array<{
+    provider_id: string;
+    guild_id?: string;
+  }>;
+}
+
+/**
+ * TODO(z):
+ * - name defaults to discord name.
+ * - admin defaults to current user.
+ * - investigate adding token stratgies.
+ * - add url to add install bot link.
+ */
+const CreateCommunity: NextPage = () => {
+  const { data: session } = useSession();
+  const [servers, setServers] = useState<Array<any>>();
+  const [noBot, setNoBot] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+  const toast = useToast();
+
+  useEffect(() => {
+    discordAxios(session?.accessToken as string).get(
+      '/users/@me/guilds'
+    ).then((res) => {
+      setServers(res.data);
+    });
+  }, [session?.accessToken]);
+
+  const onChangeFetchChannels = async (event: any) => {
+    try {
+      setNoBot(false);
+      setLoading(true);
+      const ADMIN_PERM = 2147483647;
+
+      if (event.target.value === '---') {
+        setError('client_config.0.guild_id', {
+          type: 'required',
+          message: 'Server is required',
+        });
+        setValue('name', '');
+        return;
+      }
+      
+      const server = servers?.find((server) => server.id === event.target.value);    
+
+      setValue('name', server.name);
+
+      if (server.permissions !== ADMIN_PERM) {
+        setError('client_config.0.guild_id', {
+          type: 'validate',
+          message: 'User is not an administrator of this server',
+        });
+        return;
+      }
+      const res = await privateBaseFetcher(
+        `/client/discord/${event.target.value}/channels/${session?.discordId}`
+      );
+      return res.data;
+    } catch(e) {
+      setNoBot(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const {
+    handleSubmit,
+    formState: {
+      isValid,
+      isSubmitting,
+      errors
+    },
+    register,
+    control,
+    setError,
+    setValue,
+    getValues
+  } = useForm<CommunityForm>({
```

Comment:
Getting an issue when selecting options where old validation error persists.
1. Select a server (get not admin error)
2. Select null (get must-select error)
3. Select a server (must-select error still shows)

This concept persists in various combinations. I'd advise that on-select, wipe all validation errors.

Reply:
Will update.